src/App.tsx

```diff
@@ -1,24 +1,189 @@
+    } finally {
+      setTempTodo(null);
+    }
+  };
+
+  const removeTodo = async (todoId: number) => {
+    try {
+      await deleteTodo(todoId);
+      setTodos(prevTodos => prevTodos.filter(todo => todo.id !== todoId));
+    } catch {
+      setErrorMessage('Unable to delete a todo');
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    createTodo(todoTitle);
+    setTodoTitle('');
+  };
+
+  const handleTodoTitle = (event: React.ChangeEvent<HTMLInputElement>) => (
+    setTodoTitle(event.target.value)
+  );
+
+  const handleClearCompleted = () => {
+    completedTodos.forEach(async (todo) => {
+      await removeTodo(todo.id);
+    });
+  };
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
+            type="button"
+            className={cn('todoapp__toggle-all',
+              { active: uncompletedTodos.length !== 0 })}
+          />
+
+          <form onSubmit={handleSubmit}>
+            <input
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              value={todoTitle}
+              onChange={handleTodoTitle}
+              disabled={tempTodo !== null}
+            />
+          </form>
+        </header>
+
+        <TodoList todos={filteredTodos()} removeTodo={removeTodo} />
+
+        {todos.length !== 0
+        && (
+          <footer className="todoapp__footer">
+            <span className="todo-count">
+              {`${uncompletedTodos.length} items left`}
+            </span>
+
+            <nav className="filter">
+              <a
+                href="#/"
+                className={cn('filter__link',
+                  { selected: filter === Filters.ALL })}
+                onClick={() => setFilter(Filters.ALL)}
+              >
+                All
+              </a>
+
+              <a
+                href="#/active"
+                className={cn('filter__link',
+                  { selected: filter === Filters.ACTIVE })}
+                onClick={() => setFilter(Filters.ACTIVE)}
+              >
+                Active
+              </a>
+
+              <a
+                href="#/completed"
+                className={cn('filter__link',
+                  { selected: filter === Filters.COMPLETED })}
+                onClick={() => setFilter(Filters.COMPLETED)}
+              >
+                Completed
+              </a>
```

Comment:
You can render this links using {Object.entries(Filtering).map(([key, value]) => (<a...>...)